prometheus/histogram.go

```diff
@@ -821,6 +842,13 @@ func (h *histogram) Write(out *dto.Metric) error {
 				Length: proto.Uint32(0),
 			}}
 		}
+
+		if cap(h.nativeExemplars.exemplars) > 0 {
```

Comment:
This might be taken as a mistake, so let's have a quick commentary on why `cap` is used not `len` here, WDYT?

Reply:
We use `cap` here, because we first define that, if users use a negative `NativeHistogramMaxExemplars` value, then no exemplars will be cached. (You can see this in the comment of `NativeHistogramMaxExemplars`).

So when we call `makeNativeExemplars` to initialize a new `nativeExemplars`, if `NativeHistogramMaxExemplars` is negative, we just create a slice of exemplars with cap 0 to indicates that no exemplars will be cached.

So when `Write` is called, we use cap to check if user wants to cache exemplars. If they wants, then we will output the cached exemplars. 

But I think your idea is better, because when len is 0, although cap maybe greater than 0, we could still skip the following append operation since there are no cached exemplars at that time.

@bwplotka

Reply:
If exemplars are configured, and we append the first exemplar, the `cap` is > 1, but the `len` is still 0. So it is correct and actually required to test for `cap` here.

Reply:
From Bartek's comment I believe he agrees with the usage of `cap`, he just requested a comment on top so we can remember the reason in the future

Reply:
@ArthurSens I think you are right :)
I misunderstood @bwplotka ‘s comments.

Reply:
I added a quick comment.